corehq/apps/app_manager/views/download.py

```diff
@@ -223,7 +229,16 @@ def resolve_path(path):
             path=full_path,
         ))
         if not obj.is_cached():
-            payload = request.app.fetch_attachment(full_path)
+            #lazily create language profiles to avoid slowing initial build
+            try:
+                payload = request.app.fetch_attachment(full_path)
+            except ResourceNotFound:
+                if build_profile in request.app.build_profiles and build_profile_access:
+                    request.app.create_build_files(save=True, build_profile_id=build_profile)
+                    request.app.save()
```

Comment:
Will this always (re-)create the build files (even if they already exist) if a path is requested that does not (and never will) exist in the build files?

It seems like we might want to find another way to trigger lazy creation of build files. For example, if a build profile is specified, check if its files have been built by looking for a known key like `files/{build_profile}/profile.xml` in the app attachments dict (`blobs` once migrated to blob db), and build them before trying to fetch the attachment if that key does not exist.